File: AppForms/AppForms.Web/Modules/ServerTypes/AppForms/SpecialitiesForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface SpecialitiesForm {
    Name: StringEditor;
}

export class SpecialitiesForm extends PrefixedContext {
    static formKey = 'AppForms.Specialities';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SpecialitiesForm.init)  {
            SpecialitiesForm.init = true;

            var w0 = StringEditor;

            initFormType(SpecialitiesForm, [
                'Name', w0
            ]);
        }
    }
}

File: AppForms/AppForms.Web/Modules/AppForms/Specialities/SpecialitiesForm.cs
using Serenity.ComponentModel;

namespace AppForms.AppForms.Forms;

[FormScript("AppForms.Specialities")]
[BasedOnRow(typeof(SpecialitiesRow), CheckNames = true)]
public class SpecialitiesForm
{
    public string Name { get; set; }
}